### PR TITLE
Force search form submission on enter key.

### PR DESCRIPTION
**Pull request by isoos**
#7813

### CODE REVIEW

**Comment by sigurdm** [comment]
Could you deploy this on staging to play with?

**Comment by isoos** [comment]
> Could you deploy this on staging to play with?

Deployed to staging, I think it should be online now.

**Comment by sigurdm** [comment]
I must admit I don't understand why this is better than relying on the default behavior? (I see no differences in Chrome)

Do we know which browsers have problems with the default behavior?

Could you link to the relevant discussions you found.

**Comment by isoos** [comment]
> (I see no differences in Chrome)

That is to be expected, also in Firefox, they both work as expected (by default, not sure how that changes with extensions).

> Do we know which browsers have problems with the default behavior?

This was reported through this issue: #7813. Do you recall the source?

**Comment by sigurdm** [comment]
> This was reported through this issue: https://github.com/dart-lang/pub-dev/issues/7813. Do you recall the source?

Yeah the source is a support email. Though I quoted the entirety of the email.

Maybe we do nothing until we can confirm this problem is real?

**Comment by sigurdm** [comment]
I thought the suggestion of making the icon clickable made some sense.

However checking out the google.com search bar, their icon does not seem clickable...

File: pkg/web_app/lib/src/search.dart [hunk 1]
```diff
@@ -10,12 +10,32 @@ import 'gtm_js.dart';
...
+void _setFormSubmitKeyboardHandler() {
+  final inputElem = document.querySelector('form.search-bar input[name="q"]');
+  if (inputElem != null && inputElem is InputElement) {
+    final form = inputElem.form;
+    if (form == null) {
+      return;
+    }
+    inputElem.onKeyDown.listen((e) {
+      // only handle `enter` key
+      if (e.keyCode != 13) {
```
Comment (jonasfj) [inline pkg/web_app/lib/src/search.dart:29]: Don't submit on shift+enter

**Review by jonasfj — COMMENTED** [review]
I don't mind this, but do we need it?

What browser are we having issues in?

**Comment by isoos** [comment]
> I don't mind this, but do we need it?

I think it is important to note that by default the major browsers with their default settings have the expected behavior and we wouldn't need this, unless we want to handle special configurations.

**Comment by sigurdm** [comment]
I'd say we don't land this - reimplementing default behavior seems odd - especially if we don't have clarity of which browsers have the problem, and if the problem is indeed solved by this.

**Pull request closed without merging**